etc/config/ruby.amazon.properties

```diff
@@ -1,19 +1,23 @@
 compilers=&ruby
-defaultCompiler=ruby322
+defaultCompiler=ruby33
 
-group.ruby.compilers=ruby322:ruby302:ruby274:ruby268:ruby259
+group.ruby.compilers=ruby33:ruby32:ruby31:ruby30:ruby27:ruby26:ruby25
 group.ruby.isSemVer=true
 group.ruby.baseName=Ruby
 group.ruby.groupName=Ruby YARV
 group.ruby.compilerType=ruby
 
-compiler.ruby322.semver=3.2.2
-compiler.ruby322.exe=/opt/compiler-explorer/ruby-3.2.2/bin/ruby
-compiler.ruby302.semver=3.0.2
-compiler.ruby302.exe=/opt/compiler-explorer/ruby-3.0.2/bin/ruby
-compiler.ruby274.semver=2.7.4
-compiler.ruby274.exe=/opt/compiler-explorer/ruby-2.7.4/bin/ruby
-compiler.ruby268.semver=2.6.8
-compiler.ruby268.exe=/opt/compiler-explorer/ruby-2.6.8/bin/ruby
-compiler.ruby259.semver=2.5.9
-compiler.ruby259.exe=/opt/compiler-explorer/ruby-2.5.9/bin/ruby
+compiler.ruby33.semver=3.3.4
+compiler.ruby33.exe=/opt/compiler-explorer/ruby-3.3.4/bin/ruby
+compiler.ruby32.semver=3.2.4
+compiler.ruby32.exe=/opt/compiler-explorer/ruby-3.2.4/bin/ruby
+compiler.ruby31.semver=3.1.6
+compiler.ruby31.exe=/opt/compiler-explorer/ruby-3.1.6/bin/ruby
+compiler.ruby30.semver=3.0.7
+compiler.ruby30.exe=/opt/compiler-explorer/ruby-3.0.7/bin/ruby
+compiler.ruby27.semver=2.7.8
+compiler.ruby27.exe=/opt/compiler-explorer/ruby-2.7.8/bin/ruby
```

Comment:
will need `compiler.ruby27.alias=ruby274`

Reply:
explain? I intentionally shortened all of them so they don't need to change for patch bumps.

Reply:
or: what is `ruby274` referring to in this context?

Reply:
aliases will preserve functionality of existing links pointing to the old compilers

Reply:
but... where is this existing `ruby274` you're having me reference?

OR: I'm confused and out of my depth... feel free to throw a commit on

Reply:
It's in the code you deleted. In shared links we store compilers are referred to with ID's like `ruby274`, and the only 2 ways for us to know what that is - is by normal ids like `compiler.ruby74...` And by things that are in the `.alias` property

Reply:
@zenspider is that clear enough? The IDs are exposed to users and so changing them is a "URL break". If we want to change them we can set an `alias` to treat the old ID like the new one.

Reply:
I guess it is? I still don't think I grok but that's NBD... Again, feel free to throw a commit on this branch if you want.

Would it be better if I did NOT shorten to `rubyXY` instead of `rubyXYZ`? I thought I was making this more maintainable.

Reply:
When you make a short link in Compiler Explorer, we save the state of your browser. Part of that state is the compilers you're using. We need a unique, durable identifier of the compiler(s) you've selected. The "id" (the bit in `compiler.THISID.blah`) is used as the unique identifier. That means if we change the meaning of an ID, then we break anyone who has saved a short link, or made a bookmark, or is relying on the external API to mean "I need this particular compiler".

Hopefully that helps explain a bit more why we like to keep the IDs stable. Other than keeping them "the same", there's not really any important distinction: though to avoid this kind of thing we tend to use the longest sequence of numbers that covers the salient parts of the version. That's why we used `ruby322` and similar before, it's unambiguous. Ideally we'd stick with `rubyXYZ` for ruby X.Y.Z as then there's fewer issues down the line.